### ISSUE TITLE
[Flutter Web] Browser Compatibility Issue on JavaScript "Nullish coalescing operator (??)"

### ISSUE THREAD

**Issue by xuemind** [flutter/flutter#86213]
## Description
In web app built by flutter, the generated JavaScript adopted "Nullish coalescing operator (??)" which is not supported by old version browser. So, it cause blank page with JavaScript error.

In generated ```index.html```
``` javascript
waitForActivation(reg.installing ?? reg.waiting);
```
it cause the following error in chrome version less than 80.
```
VM119:1 Uncaught SyntaxError: Unexpected token ?
```
Please check the compatible browser list here:  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator#browser_compatibility
This operator only supported since Chrome version 80. But there are still lots of existing browser cannot meet this requirement.

## Steps to Reproduce
```
flutter create sampleweb
cd sampleweb
flutter run -d chrome
```
**Access the website with old version of chrome < version 80**

**Expected results:** 
Expect the web page rendered in browser.

**Actual results:** 
Nothing displayed in mobile browser.
Check the browser console with developer tools and get the following error message. 
```
VM119:1 Uncaught SyntaxError: Unexpected token ?
```

**Environment**
```
❯ flutter channel
Flutter channels:
  master
  dev
  beta
* stable
❯ flutter --version
Flutter 2.2.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (8 days ago) • 2021-07-01 12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4
```

<details>
  <summary>Logs</summary>

```
```

```
```

</details>

**Comment by xuemind** [comment]
is it possible to generate JavaScript for web without "Nullish coalescing operator"? It is easy to replace it with much normal expression. The syntactic sugar is used to improve developer productivity. But it not good idea for generated language which may cause compatibility issue and reduce readability.

**Comment by xuemind** [comment]
I manually replaced ```??``` with ```||``` operator and works on old chrome v76. Not sure if it can cover edge cases, but at least it works.
```
waitForActivation(reg.installing || reg.waiting);
```

**Comment by xuemind** [comment]
check the with source code. it is same issue with #83176 and fixed in #83732. But not in stable channel yet.
